src/main/java/com/example/libraryms/controllers/LoginController.java

```diff
@@ -0,0 +1,74 @@
+package com.example.libraryms.controllers;
+
+import com.example.libraryms.services.LoginService;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.control.PasswordField;
+import javafx.scene.control.TextField;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.scene.Parent;
+import javafx.fxml.FXMLLoader;
+import java.io.IOException;
+
+public class LoginController {
+
+    @FXML
+    private TextField usernameField;
+
+    @FXML
+    private PasswordField passwordField;
+
+    @FXML
+    private Label errorLabel;
+
+    private LoginService loginService = new LoginService();
+
+    @FXML
+    public void handleLogin() {
+        String username = usernameField.getText();
+        String password = passwordField.getText();
+
+        // Basic validation
+        if (username.isEmpty() || password.isEmpty()) {
+            errorLabel.setText("Both fields are required.");
+            return;
+        }
+
+        boolean isValidUser = loginService.authenticate(username, password);
+
+        if (isValidUser) {
+            System.out.println("Login successful");
+            loadDashboard();
+        } else {
+            errorLabel.setText("Invalid username or password.");
+        }
+    }
+
+
+    private void loadDashboard() {
+        try {
+            FXMLLoader loader = new FXMLLoader(getClass().getResource("/com/example/libraryms/dashboard.fxml"));
+            Parent root = loader.load();
+            Stage stage = (Stage) usernameField.getScene().getWindow();
+            stage.setScene(new Scene(root));
+            stage.show();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    @FXML
+    public void theSignup() {
+        try {
+            FXMLLoader loader = new FXMLLoader(getClass().getResource("/com/example/libraryms/sign-up.fxml"));
+            Parent root = loader.load();
+            Stage stage = (Stage) usernameField.getScene().getWindow();
+            stage.setScene(new Scene(root));
+            stage.show();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
```

Comment:
Stack traces are used for development purposes. don't push them to live environments, instead handle the exception

eg.
```
throw new RuntimeException(e)
```